snapcraft.yaml

```diff
@@ -29,9 +25,25 @@ passthrough:
 # https://forum.snapcraft.io/t/5547/10
 parts:
   snapd:
-    # FIXME: this should probably go upstream
-    plugin: x-builddeb
+    plugin: nil
     source: .
+    build-snaps: [go/1.10/stable]
+    override-build: |
+      # set version 
```

Comment:
You could perhaps split this block into override-pull and override-build.  Everything down down to, and including, get-deps could appropriately be done in the pull step.  But I believe the net effect is precisely the same in this case, so it's really just a matter of preference.

Reply:
Thanks, I ended up splitting this up, but I had to put `set-version` at the end of override-pull, because it needs tools installed by the `apt-get build-dep` call to generate a valid version string.